Fix compilation failure on PowerPC:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/GenericPacketMath.h:862:67: error: no matching function for call to 'atan(const __vector(2) double&)'

Add the upstream commit https://gitlab.com/libeigen/eigen/-/commit/886aad136111eeeb7604e1d17f62efcc4d824568

Author: Alexander Grund (TU Dresden)

diff --git a/third_party/eigen3/disable-atan-on-ppc.patch b/third_party/eigen3/disable-atan-on-ppc.patch
new file mode 100644
index 00000000000..4614799b66a
--- /dev/null
+++ b/third_party/eigen3/disable-atan-on-ppc.patch
@@ -0,0 +1,13 @@
+diff --git a/Eigen/src/Core/arch/AltiVec/PacketMath.h b/Eigen/src/Core/arch/AltiVec/PacketMath.h
+index 37398de15..d9ddb5e35 100644
+--- a/Eigen/src/Core/arch/AltiVec/PacketMath.h
++++ b/Eigen/src/Core/arch/AltiVec/PacketMath.h
+@@ -2708,7 +2708,7 @@ template<> struct packet_traits<double> : default_packet_traits
+     HasAbs  = 1,
+     HasSin  = 0,
+     HasCos  = 0,
+-    HasATan = 1,
++    HasATan = 0,
+     HasLog  = 0,
+     HasExp  = 1,
+     HasSqrt = 1,
diff --git a/third_party/eigen3/workspace.bzl b/third_party/eigen3/workspace.bzl
index 91f471c3def..9fa96a38310 100644
--- a/third_party/eigen3/workspace.bzl
+++ b/third_party/eigen3/workspace.bzl
@@ -14,6 +14,7 @@ def repo():
     tf_http_archive(
         name = "eigen_archive",
         build_file = "//third_party/eigen3:eigen_archive.BUILD",
+        patch_file = ["//third_party/eigen3:disable-atan-on-ppc.patch"],
         sha256 = EIGEN_SHA256,
         strip_prefix = "eigen-{commit}".format(commit = EIGEN_COMMIT),
         urls = tf_mirror_urls("https://gitlab.com/libeigen/eigen/-/archive/{commit}/eigen-{commit}.tar.gz".format(commit = EIGEN_COMMIT)),
